feat: transform recommendation to guidance in layer2 OSCAL generation

zohayb23 opened the pull request:
## Description
Aligned layer2's OSCAL generation with layer1's implementation by transforming "recommendation" to "guidance". 

## Changes Made
- Changed part name from "recommendation" to "guidance" to match layer1's implementation
- Updated ID format to match layer1's format: `{controlId}_smt.{partId}_gdn`

## Reference
Original layer1 implementation: https://github.com/ossf/gemara/blob/609b7a07b81e5a84cde8eb620f2885906ead80ba/layer1/oscal_generator.go#L286-L295

jpower432 commented on layer2/oscal_generator.go:
```diff
@@ -62,8 +62,8 @@ func (c *Catalog) ToOSCAL(controlHREF string) (oscal.Catalog, error) {
-							Name:  "recommendation",
+							ID:    fmt.Sprintf("%s_smt.%s_gdn", control.Id, ar.Id),
+							Name:  "guidance",
 							Ns:    oscalUtils.GemaraNamespace,
```
The namespace can be removed as well since `guidance` is defined under the default OSCAL namespace.

jmeridth commented:
based on https://pages.nist.gov/OSCAL/learn/concepts/layer/control/catalog/sp800-53rev5-example/#parts `_smt` is for statements ([link](https://pages.nist.gov/OSCAL/learn/concepts/layer/control/catalog/sp800-53rev5-example/#parts:~:text=%3Cpart%20name%3D%22statement%22%20id%3D%22ac%2D1_smt%22%3E)). The `_gdn` suffix is for guidance ([link](https://pages.nist.gov/OSCAL/learn/concepts/layer/control/catalog/sp800-53rev5-example/#parts:~:text=%3Cpart%20name%3D%22guidance%22%20id%3D%22ac%2D1_gdn%22%3E)). Am I wrong? Can they be daisy-chained like this?

jpower432 commented:
> If a particular way of structuring the information is necessary, it is important to note that controls can have sub-controls, each with its own statement , assessment-objective and assessment-method. The assessment report can be done with any desired granularity. The control and sub-control approach will allow you to use core OSCAL and have the structure @eddie-knight expects.

Thanks for suggesting sub-controls @iMichaela. That should also be helpful when we want to make OSCAL Profiles to enable us to include or exclude `AssessmentRequirements` based on applicabilty.

zohayb23 commented:
@jpower432 @iMichaela I've updated the code based on your feedback with a new commit that:

1. Implements proper OSCAL structure:
   - Moved guidance to top-level (no nesting)
   - Uses sub-controls for assessment requirements
   - Fixed control class to use `family.Id` instead of `Title`
   - Removed unnecessary namespace for guidance parts

2. Added proper linking:
   - Assessment objectives link back to their statements using `rel="assessment-for"`
   - Each sub-control has its own statement, guidance, and assessment parts

jpower432 commented on layer2/oscal_generator.go:
```diff
@@ -53,43 +53,56 @@ func (c *Catalog) ToOSCAL(controlHREF string) (oscal.Catalog, error) {
+			var subControls []oscal.Control
+			for _, ar := range control.AssessmentRequirements {
+				subControl := oscal.Control{
+					ID:    fmt.Sprintf("%s.%s", control.Id, ar.Id),
```
I'm thinking the `ar.Id`  should just be used here and ID locations. By convention, the Assessment Requirement IDs are already expressed in this format. The outcome sub-control ID for this change is, for example, `OSPS-AC-02.OSPS-AC-02.01`
```suggestion
					ID:    ar.Id,
```

jpower432 commented:
Thanks for making the changes @zohayb23. I thought this was a good opportunity to test the validation process in #145. I followed in the steps from that draft process and used it to suggested some changes in a [PR](https://github.com/zohayb23/gemara/pull/1) to your branch.

jpower432 reviewed (approved):
LGTM
